docs/DeveloperGuide.adoc

```diff
@@ -170,168 +216,116 @@ Classes used by multiple components are in the `tatracker.commons` package.
 name.
 
-5. The AddModuleCommandParser then creates an AddModuleCommand object and passes
-it the created module. The parser then returns the AddModuleCommand
+5. The `AddModuleCommandParser` then creates an `AddModuleCommand` object and passes
+it the created module. The parser then returns the `AddModuleCommand` object.
 
-6. LogicManager calls AddModuleCommad's execute method. The AddModuleCommand object
+6. LogicManager calls `AddModuleCommand` 's execute method. The `AddModuleCommand` object
 checks whether a module with the given module code already exists in TA-Tracker.
 If it does, a command exception is thrown saying that a module with the given module
 code already exists in the TA-Tracker.
 
 7. If no such module exists, the module is added to the TA-Tracker.
 
-8. The SortGroupCommand returns a CommandResult with a success message.
+8. The `AddGroupCommand` returns a `CommandResult` with a success message.
 
-The command used to delete a module has been implemented in a similar way. Tha main
-difference is that the DeleteModuleCommand checks whether an object with the given
+The command used to delete a module has been implemented in a similar way. The main
+difference is that the `DeleteModuleCommand` checks whether an object with the given
 module code exists in the TA-Tracker. If no such module exists, a command exception
 is thrown saying that a module with the given module code doesn't exist. If it does
 exist, first all the sessions linked to that module are removed.
 Then the module is removed from the TA-Tracker.
 
-==== Implementation of the Group Add and Delete Commands
+The `module edit` command has been implemented in a similar manner.
 
-The following activity diagram shows the steps taken by the AddGroupCommand object
-when the execute method is called.
+==== Implementation of the Group Add, Delete and Edit Commands
 
-.Group Add - Activity Diagram
-image::AddGroupActivityDiagram.png[]
+A group is added to the TA-Tracker in a similar manner to to how a module is added to
+the TA-Tracker.
 
-It should be noted that these are the steps followed assuming that no exception is
-thrown. Before getting the module from TA-Tracker's model, the DeleteGroupCommand object
-checks whether such a module even exists. If it doesn't exists, it throws a command
-exception saying that no such module exists. Before adding a group to the module, the
-object even checks whether the module already has a group with the given group code.
-If it exists, a command exception is thrown saying that there is already a group
-with the given group code.
+The following steps are taken once the execute command of an AddGroupCommand object
+is called:
+
+. The AddGroupCommand object checks whether the module is present in the model of the TA-Tracker.
+.. If it is not present, an exception is thrown explaining that the module doesn't
+exist.
+. If the module does exist, the module is retrieved from the TA-Tracker's model.
```

Comment:
Perhaps this could be written like this?

1. Does `module` exist?
* YES: Retrieve the `module`
* NO: Throw an `InvalidModuleException`

2. Does `group` exist in `module`?
* YES: Add the `group` into the` module`, and display a `SUCCESS` message
* NO: Throw a `DuplicateGroupException`